src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,37 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.LinkedList;
+import java.util.List;
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends T> machines) {
+        if (machines.equals(Bulldozer.class)) {
+            return new LinkedList<>(new BulldozerProducer().get());
+        } else if (machines.equals(Excavator.class)) {
```

Comment:
should we use `else` after `return` statements in the if block

Reply:
in this case i think if-else if, as same as if-if